File: concert-ticketing/src/main/java/com/hhplus/concertticketing/common/config/AsyncConfig.java
package com.hhplus.concertticketing.common.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.scheduling.annotation.EnableAsync;
import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;

import java.util.concurrent.Executor;

@Configuration
@EnableAsync
public class AsyncConfig {
    @Bean(name = "taskExecutor")
    public Executor asyncExecutor() {
        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
        executor.setCorePoolSize(2); // 기본 스레드 수
        executor.setMaxPoolSize(100); // 최대 스레드 수
        executor.setQueueCapacity(5); // 큐 용량
        executor.setThreadNamePrefix("Async-"); // 스레드 이름 접두사
        executor.initialize();
        return executor;
    }
}

File: concert-ticketing/src/main/java/com/hhplus/concertticketing/common/config/AsyncConfig.kt
package com.hhplus.concertticketing.common.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.annotation.EnableAsync
import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor
import java.util.concurrent.Executor

/**
 * 비동기 처리를 위한 스레드풀 설정
 */
@Configuration
@EnableAsync
class AsyncConfig {
    @Bean(name = "taskExecutor")
    fun asyncExecutor(): Executor {
        val executor = ThreadPoolTaskExecutor()
        executor.corePoolSize = 2 // 기본 스레드 수
        executor.maxPoolSize = 100 // 최대 스레드 수
        executor.setQueueCapacity(5) // 큐 용량
        executor.threadNamePrefix = "Async-" // 스레드 이름 접두사
        executor.initialize()
        return executor
    }
}
